packages/contracts-bedrock/src/L2/GasPriceOracle.sol

```diff
@@ -84,16 +84,86 @@ contract GasPriceOracle is ISemver {
     /// @param _data Unsigned fully RLP-encoded transaction to get the L1 gas for.
     /// @return Amount of L1 gas used to publish the transaction.
     function getL1GasUsed(bytes memory _data) public view returns (uint256) {
-        uint256 total = 0;
-        uint256 length = _data.length;
-        for (uint256 i = 0; i < length; i++) {
-            if (_data[i] == 0) {
-                total += 4;
-            } else {
-                total += 16;
-            }
-        }
+        // 0s in compressed data are rare, so we simply multiply by 16 to estimate calldata costs.
+        uint256 total = flzCompressLen(_data) * 16;
         uint256 unsigned = total + overhead();
         return unsigned + (68 * 16);
     }
+
+    /// @notice Version of https://github.com/Vectorized/solady/blob/main/src/utils/LibZip.sol
+    ///         that only returns the length of the data if it were to be compressed. This saves
+    ///         gas over actually compressing the data, given we only need the length.
+    /// @dev Returns the length of the compressed data.
+    /// @custom:attribution Solady <https://github.com/Vectorized/solady>
+    function flzCompressLen(bytes memory data) internal pure returns (uint256 n) {
```

Comment:
Can you put this in a library inside of `src/libraries`? Maybe `Compression.sol` or something like that. Then we need to differentially test against the Go version. It would be easier to do so by having the Go code in the monorepo

Reply:
@tynes Sure, i'll move it to a separate file in `libraries`.

I don't think op-geth has any dependencies on the monorepo yet, to avoid circular deps; but we can differentially test it by pulling in the op-geth commit here.